src/flp.rs

```diff
@@ -304,7 +304,8 @@ pub trait Type: Sized + Eq + Clone + Debug {
             // evaluation of the gadget.
             let m = wire_poly_len(gadget.calls());
             let m_inv =
-                Self::Field::from(<Self::Field as FieldElement>::Integer::try_from(m).unwrap())
+                Self::Field::try_from(<Self::Field as FieldElement>::Integer::try_from(m).unwrap())
+                    .unwrap()
```

Comment:
I'm wary of adding more `unwrap()` to non-test code. We have `FlpError::Field(FieldError)` so I think we should propagate the error here.
```suggestion
                Self::Field::try_from(<Self::Field as FieldElement>::Integer::try_from(m).unwrap())?
```